src/game/client/components/items.cpp

```diff
@@ -301,19 +289,36 @@ void CItems::RenderLaser(const CLaserData *pCurrent, bool IsPredicted)
 		Alpha = g_Config.m_ClShowOthersAlpha / 100.0f;
 	}
 
+	const ColorRGBA OuterColor = color_cast<ColorRGBA>(ColorHSLA(ColorOut).WithAlpha(Alpha));
+	const ColorRGBA InnerColor = color_cast<ColorRGBA>(ColorHSLA(ColorIn).WithAlpha(Alpha));
+
+	float Ticks;
+	if(Type == LASERTYPE_DOOR)
+	{
+		Ticks = 1.0f;
+	}
+	else if(IsPredicted)
+	{
+		int PredictionTick = Client()->GetPredictionTick();
+		Ticks = (float)(PredictionTick - pCurrent->m_StartTick) + Client()->PredIntraGameTick(g_Config.m_ClDummy);
+	}
+	else
+		Ticks = (float)(Client()->GameTick(g_Config.m_ClDummy) - pCurrent->m_StartTick) + Client()->IntraGameTick(g_Config.m_ClDummy);
+
+	RenderLaser(pCurrent->m_From, pCurrent->m_To, OuterColor, InnerColor, Ticks);
+}
+
+void CItems::RenderLaser(const vec2 &From, const vec2 &Pos, ColorRGBA &OuterColor, ColorRGBA &InnerColor, float Ticks)
```

Comment:
Doesn't necessarily improve readability to me to split it either.

Pass the arguments by value instead of by reference because they are small if you want a separate function:

```suggestion
void CItems::RenderLaser(vec2 From, vec2 Pos, ColorRGBA OuterColor, ColorRGBA InnerColor, float Ticks)
```